### PR TITLE
[patch] INTENG-17048 change the OG tags set logic for link data.

### PR DESCRIPTION
**Pull request by hana-branch**
# Pull Request Template

## Description
Change not to set the OG keys (eg: `$og_title`, `$og_description`, `$og_image_url`, `$og_video`, `$og_type`) that are gotten from `<meta>` tags in the webpage as the link data when there are `$og_redirect`, `$desktop_url`, or `$fallback_url` 
So we can scrap the OG tag from the webpage from `$og_redirect`, `$desktop_url`, or `$fallback_url`  

Fixes # https://branch.atlassian.net/browse/INTENG-17048

## Type of change

Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [X] Integration test

## JS Budget Check

Please mention the size in kb before abd after this PR

| Files            | Before      | After       | 
| -----------      | ----------- | ----------- |
| dist/build.js.   |    164 KB        |    74 KB       |
| dist/build.min.js|  164 KB          |       74 KB    |

## Mentions: 
List the person or team responsible for reviewing proposed changes.

### CODE REVIEW

File: src/1_utils.js
Comment (JagadeeshKaricherla-branch) [inline src/1_utils.js:353]: we can avoid evaluating the condition with every setOgTags call.

`var hasOGRedirectwithFallback = linkData['$og_redirect'] && linkData['$fallback_url'] && linkData['$desktop_url'];`

and use it

`data['$og_description'] = hasOGRedirectwithFallback ? utils.getOpenGraphContent('description') : null;`

And remove the new setOgTags method.

Comment (hana-branch) [reply src/1_utils.js]: Thanks for your advice, I just changed the code based on your feedback.

**Review by JagadeeshKaricherla-branch — APPROVED** [review]
LGTM

**Comment by hana-branch** [comment]
@JagadeeshKaricherla-branch To confirm, I just need to squash and merge this change once I get the approval? Thanks

**Pull request merged**
